quinn-proto/src/connection/streams.rs

```diff
@@ -332,6 +332,13 @@ impl Streams {
         Ok(())
     }
 
+    /// Check if the peer has finished sending data
+    pub fn is_peer_finished(&self, id: StreamId) -> Result<bool, UnknownStream> {
+        let stream = self.recv.get(&id).ok_or(UnknownStream { _private: () })?;
+        // Only bother if there's data we haven't received yet
```

Comment:
This comment makes more sense at the callsite.

Reply:
Fixed.